### PR TITLE
fix(web): resolve unhandled exception in promise

### LINKED ISSUE #5353: bug(web): Uncaught (in promise) Error
Found while testing #5260

For testing the error handling of kmwkeyboards.keymanCloudRequest(), I intentionally defined an invalid URL
```javascript
 const URL='https://apis.keyman.com/cloud/4.0/'
                + ((arguments.length > 1) && byLanguage ? 'languages' : 'keyboards');
```

This caused the following console exception
> Uncaught (in promise) Error: Could not find a keyboard with that ID.
    at HTMLScriptElement.Lscript.onerror (kmwkeyboards.ts:1258)

Turns out this also can be repro'd on the current `master` branch (as of 15.0.73 alpha) after making the same change to URL, separate from my Promise API PR's.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #5353.

I made the promise usage consistent in the relevant addKeyboards functions, instead of mixing promise and async/await style code, which made it easier to spot the place(s) where promises had not been returned as we needed.

# User Testing

* TEST_PROMISE: make sure that existing keyboards load correctly, e.g. on testing/unminified, from cloud (e.g. Swedish is a cloud keyboard).
* TEST_PROMISE_FAILS: add an entry to your hosts file for `127.0.0.0 api.keyman.com` temporarily, to force keyboard loading to fail. Then reload the above page, with Web Developer tools open, and verify that no 'uncaught in promise' error is shown. (Don't forget to remove the entry from your hosts file again afterwards.)

### CODE REVIEW

File: web/testing/commonHeader.js [hunk 1]
```diff
@@ -98,16 +128,16 @@
...
       case 2:
         sKbd=document.getElementById('kbd_id2').value.toLowerCase();
-        kmw.addKeyboards('@'+sKbd);
+        doAddKeyboards('@'+sKbd);
         break;
       case 3:
         // Add keyboard for comma-separated language name(s)
         sKbd=document.getElementById('kbd_id3').value;
-        kmw.addKeyboardsForLanguage(sKbd);
+        doAddKeyboardsForLanguage(sKbd);
```
Comment (darcywong00) [inline web/testing/commonHeader.js:140]: Can the commonHeader.js change also be applied to
web/testing/promise-api/promisehdr.js ?

Comment (mcdurdin) [reply web/testing/commonHeader.js:140]: This looks like it is already done in promisehdr.js? Slightly differently, but the same basic idea.

**Comment by darcywong00** [comment]
* **TEST_PROMISE** - PASSED
The test page loads the keyboards

* **TEST_PROMISE_FAILS** - PASSED
No uncaught promises in the console

**Review by darcywong00 — APPROVED** [review]
lgtm - will need to resolve 1 merge conflict

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.153-alpha](https://keyman.com/downloads/releases/alpha/15.0.153)

**Pull request merged**